src/components/LibraryContainer.tsx

```diff
@@ -25,20 +23,85 @@ export interface LibraryContainerStates {
+
+        // Set handlers after methods are bound.
+        this.props.libraryController.setLoadedTypesJsonHandler = this.setLoadedTypesJson;
+        this.props.libraryController.setLayoutSpecsJsonHandler = this.setLayoutSpecsJson;
+        this.props.libraryController.refreshLibraryViewHandler = this.refreshLibraryView;
+
         this.state = { inSearchMode: false };
-        this.generatedSections = buildLibrarySectionsFromLayoutSpecs(this.props.loadedTypesJson, this.props.layoutSpecsJson, 
-        this.props.defaultSectionString, this.props.miscSectionString);
+    }
+
+    setLoadedTypesJson(loadedTypesJson: any, append: boolean = true): void {
+
+        if (!loadedTypesJson) {
+            throw new Error("Parameter 'loadedTypesJson' must be supplied");
+        }
+
+        if (!loadedTypesJson.loadedTypes || (!Array.isArray(loadedTypesJson.loadedTypes))) {
+            throw new Error("'loadedTypesJson.loadedTypes' must be a valid array");
+        }
+
+        // If there is no existing loadedTypesJson object, or the call
+        // is meant to replace the existing one, then assign it and bail.
+        if (!this.loadedTypesJson || (!append)) {
+            this.loadedTypesJson = loadedTypesJson;
+            return;
+        }
+
+        // To append to the existing 'loadedTypesJson.loadedTypes object' (merge both arrays).
+        Array.prototype.push.apply(this.loadedTypesJson.loadedTypes, loadedTypesJson.loadedTypes);
+    }
+
+    setLayoutSpecsJson(layoutSpecsJson: any, append: boolean = true): void {
+
+        if (!layoutSpecsJson) {
+            throw new Error("Parameter 'layoutSpecsJson' must be supplied");
+        }
+
+        if (!layoutSpecsJson.sections || (!Array.isArray(layoutSpecsJson.sections))) {
+            throw new Error("'layoutSpecsJson.sections' must be a valid array");
+        }
+
+        // If there is no existing layoutSpecsJson object, or the call
+        // is meant to replace the existing one, then assign it and bail.
+        if (!this.layoutSpecsJson || (!append)) {
+            this.layoutSpecsJson = layoutSpecsJson;
+            return;
+        }
+
+        // Otherwise, recursively replace/append each section.
+        LibraryUtilities.updateSections(this.layoutSpecsJson, layoutSpecsJson);
+    }
+
+    refreshLibraryView(): void {
+
+        this.generatedSections = LibraryUtilities.buildLibrarySectionsFromLayoutSpecs(
+            this.loadedTypesJson, this.layoutSpecsJson,
+            this.props.defaultSectionString, this.props.miscSectionString);
 
         // Obtain the categories from each section to be added into the filtering options for search
         for (let section of this.generatedSections) {
             for (let childItem of section.childItems)
                 this.searchCategories.push(childItem.text);
         }
+
+        // Just to force a refresh of UI.
+        this.setState({ inSearchMode: this.state.inSearchMode });
```

Comment:
Yes it definitely will (without this the view wasn't updated). I think `React.js` does this because state object may sometimes be object references, even when the reference does not change, its internal values may. So `setState` must trigger a render.